src/wwwroot/js/analytics.js

```diff
@@ -21,11 +40,78 @@ function scrollToAnalyticalResultDivision(){
     }, 1000);
 }
 
+function clearOnRecurrentRequests() {
+    maleFemaleRatio = { "maleOnly":0, "femaleOnly":0, "combined":0, "typeBoysHasGirls":0, "typeGirlsHasBoys":0 };
+    // Destroy chart on recurrent requests
+    if(mfDoughnutChart != null) {
+        mfDoughnutChart.destroy();
+    }
+}
+
 function toTitleCase(str) {
     return str.replace(
         /\w\S*/g,
         function (txt) {
             return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase();
         }
     );
+}
+
+function filterAPIData(data) {
+    for(var i = data.length - 1; i >=0 ; i--){
+        filterMaleFemaleRatio(data[i]["MFRatio"], data[i]["type"]);
+    }
+
+    var mfPropertiesArray = $.map(maleFemaleRatio, function (v) { return v; });
+    createDoughnutBasedOnMFRatio(Object.keys(maleFemaleRatio), mfPropertiesArray, 'doughnut', $('#_maleFemaleCombinedDoughnut'), '');
+    summarizeMFinTable(data.length);
+}
+
+function filterMaleFemaleRatio(ratio, typeOfSchool) {
```

Comment:
Now this function has brought an interesting revelation to me. I discovered that some schools were labeled co-ed or joint in the database although the MFRatio is 1.0 — some of these schools even have 'GIRLS HIGH SCHOOL' in their names!

I will fix these errors in the database as soon as I can. Then hopefully we won't have to see weird schools that are `typeBoysHasGirls` or `typeGirlsHasBoys` 🤣